### ISSUE TITLE
Menu selection lost by accidental drop

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap#2277]
I start with

![image](https://user-images.githubusercontent.com/2138696/49435143-cb7a6880-f7ad-11e8-82be-df9c72a30115.png)

I accidentally drop 'x' in the wrong place.

![image](https://user-images.githubusercontent.com/2138696/49435176-e1882900-f7ad-11e8-89db-4874ffe8d412.png)

Then when I fix it the menu item no longer shows 'line'.

![image](https://user-images.githubusercontent.com/2138696/49435222-067c9c00-f7ae-11e8-8271-e8633aaa04d9.png)

Text values are restored - shouldn't menu selections as well?

**Comment by jguille2** [comment]
No,
Moving reporters from an input  slot, Snap! always return to the default value of that slot. And here, the default value of the "split" block is " ", not "line".

**Comment by ToonTalk** [comment]
Right. I see now that the text also reverts to the default value.

So in my scenario I could have done 'undo' 3 times and then dropped it where I intended. (Not well motivated here but if there was a long text value instead.)

Maybe as an enhancement Snap! could remember the previous value when a reporter is dropped on an input slow.

**Comment by DyslexicAwe** [comment]
> Maybe as an enhancement Snap! could remember _the previous value_ when a reporter is dropped on an input slow.

Yay! This could be this Season Holiday's small present for us, faithful users.

**Comment by jmoenig** [comment]
Wait! We have *undo* for this!!! When you *undo* the drop Snap! restores the *previous* value, not the default one!!!

**Comment by ToonTalk** [comment]
Right. I drop it on the wrong input, pick it up, drop it where it belongs and then need to hit undo 3 times to fix things. A bit awkward but maybe that's just the way things are.

What is worse is that I think I was lucky I noticed the menu selection had changed. Otherwise later when I tracked down the bug I would wonder how it got to be wrong since I remember selecting "line".

**Comment by jmoenig** [comment]
No! When you drop it the first time *that's* when you hit "undo" right away! Don't pick it up first, just hit "undo" at everything will be fine!

**Comment by ToonTalk** [comment]
Yes that is the best practice but I wonder what fraction of users would do that rather than pick it up and move it.

**Comment by DyslexicAwe** [comment]
Jens is right, but he expects users to be as smart as he is, which is true only for a small fraction of users, as Ken pointed out. So... May we, the less-then-smart users, be given a small concession (or present), because these days are a bit dark and it would brighten them up a little. Please.
